controller/volume_controller.go

```diff
@@ -536,6 +536,23 @@ func (vc *VolumeController) ReconcileVolumeState(v *longhorn.Volume, e *longhorn
 		}
 	}
 
+	neverAttached := true
+	for _, r := range rs {
+		if r.Spec.HealthyAt != "" {
+			neverAttached = false
+			break
+		}
+	}
+
+	if neverAttached && v.Spec.FromBackup != "" && v.Spec.Standby == false {
+		v.Spec.RestorationRequired = true
+		usableNode, err := vc.ds.GetRandomReadyNode()
+		if err != nil {
+			return err
+		}
+		v.Spec.NodeID = usableNode.Name
+	}
+
```

Comment:
The default "RestorationRequired" field is false, and I need to set it to be true for newly restored volume. And all of these logic should be implemented in controller (i.e. not in manager when "create" is called),  so I still need a way to distinguish "newly" restored volume (the restored volume has never been attached), that's the reason for this "neverAttached" boolean.